### PR TITLE
Longhorn 1.2.3 & 1.1.3 for v2.5

### PR DESCRIPTION
**Pull request by innobead**
Signed-off-by: David Ko <dko@suse.com>

After approvals, need QA (@khushboo-rancher ) to do the final verification.

### CODE REVIEW

**Comment by innobead** [comment]
ref: https://github.com/rancher/charts/pull/1582 & https://github.com/rancher/charts/pull/1582#discussion_r735941502

**Comment by khushboo-rancher** [comment]
Validated installation, upgrade, uninstallation and  P0 cases of the charts v1.1.3-rc2 and v1.2.3-rc1 on Rancher v2.5.11 - **LGTM**

The mirror images are pending to be merged.

**Review by shuo-wu — DISMISSED** [review]
LGTM

**Review by jiaqiluo — CHANGES_REQUESTED** [review]
For both longhorn charts, can we add the following annotations to the chart.yaml of the main chart:

```
catalog.cattle.io/kube-version: 
catalog.cattle.io/rancher-version: 
catalog.cattle.io/upstream-version:
```

**Comment by jiaqiluo** [comment]
can you update the table in [this issue](https://github.com/rancher/rancher/issues/35100) to include Longhorn?

**Review by aiyengar2 — CHANGES_REQUESTED** [review]
Ran the following scripts to check diffs:

```diff
arvindiyengar: ~/Rancher/charts/src/github.com/rancher/charts 
$ diff -uNr charts/longhorn-1.1/longhorn/1.1.200+up1.1.2 charts/longhorn-1.1/longhorn/1.1.300-rc2+up1.1.3-rc2 > main-chart-1.1.diff
arvindiyengar: ~/Rancher/charts/src/github.com/rancher/charts 
$ diff -uNr charts/longhorn-1.1/longhorn-crd/1.1.200+up1.1.2 charts/longhorn-1.1/longhorn-crd/1.1.300-rc2+up1.1.3-rc2 > crd-chart-1.1.diff
arvindiyengar: ~/Rancher/charts/src/github.com/rancher/charts 
$ diff -uNr charts/longhorn-1.2/longhorn/1.2.200+up1.2.2 charts/longhorn-1.2/longhorn/1.2.300-rc1+up1.2.3-rc1 > main-chart-1-2.diff
arvindiyengar: ~/Rancher/charts/src/github.com/rancher/charts 
$ diff -uNr charts/longhorn-1.2/longhorn-crd/1.2.200+up1.2.2 charts/longhorn-1.2/longhorn-crd/1.2.300-rc1+up1.2.3-rc1 > crd-chart-1-2.diff
```

CRD diffs look fine, the main chart diffs have some issues though.

### 1.1

```diff
diff -uNr charts/longhorn-1.1/longhorn/1.1.200+up1.1.2/Chart.yaml charts/longhorn-1.1/longhorn/1.1.300-rc2+up1.1.3-rc2/Chart.yaml
--- charts/longhorn-1.1/longhorn/1.1.200+up1.1.2/Chart.yaml	2021-12-09 12:37:14.000000000 -0800
+++ charts/longhorn-1.1/longhorn/1.1.300-rc2+up1.1.3-rc2/Chart.yaml	2021-12-09 12:37:14.000000000 -0800
@@ -3,13 +3,12 @@
   catalog.cattle.io/certified: rancher
   catalog.cattle.io/display-name: Longhorn
   catalog.cattle.io/namespace: longhorn-system
-  catalog.cattle.io/rancher-version: <2.5.99-0
   catalog.cattle.io/os: linux
   catalog.cattle.io/provides-gvr: longhorn.io/v1beta1
   catalog.cattle.io/release-name: longhorn
   catalog.cattle.io/ui-component: longhorn
 apiVersion: v1
-appVersion: v1.1.2
+appVersion: v1.1.3-rc2
 description: Longhorn is a distributed block storage system for Kubernetes.
 home: https://github.com/longhorn/longhorn
 icon: https://raw.githubusercontent.com/cncf/artwork/master/projects/longhorn/icon/color/longhorn-icon-color.png
@@ -21,7 +20,7 @@
 - device
 - iscsi
 - nfs
-kubeVersion: '>=v1.16.0-r0'
+kubeVersion: '>=v1.16.0-r0 <v1.22.0-r0'
 maintainers:
 - email: maintainers@longhorn.io
   name: Longhorn maintainers
@@ -36,4 +35,4 @@
 - https://github.com/longhorn/longhorn-ui
 - https://github.com/longhorn/longhorn-tests
 - https://github.com/longhorn/backing-image-manager
-version: 1.1.200+up1.1.2
+version: 1.1.300-rc2+up1.1.3-rc2
```

### 1.2

```diff
diff -uNr charts/longhorn-1.2/longhorn/1.2.200+up1.2.2/Chart.yaml charts/longhorn-1.2/longhorn/1.2.300-rc1+up1.2.3-rc1/Chart.yaml
--- charts/longhorn-1.2/longhorn/1.2.200+up1.2.2/Chart.yaml	2021-12-09 12:37:14.000000000 -0800
+++ charts/longhorn-1.2/longhorn/1.2.300-rc1+up1.2.3-rc1/Chart.yaml	2021-12-09 12:37:14.000000000 -0800
@@ -1,5 +1,4 @@
 annotations:
-  catalog.cattle.io/rancher-version: <2.5.99-0
   catalog.cattle.io/auto-install: longhorn-crd=match
   catalog.cattle.io/certified: rancher
   catalog.cattle.io/display-name: Longhorn
@@ -9,7 +8,7 @@
   catalog.cattle.io/release-name: longhorn
   catalog.cattle.io/ui-component: longhorn
 apiVersion: v1
-appVersion: v1.2.2
+appVersion: v1.2.3-rc1
 description: Longhorn is a distributed block storage system for Kubernetes.
 home: https://github.com/longhorn/longhorn
 icon: https://raw.githubusercontent.com/cncf/artwork/master/projects/longhorn/icon/color/longhorn-icon-color.png
@@ -35,4 +34,4 @@
 - https://github.com/longhorn/longhorn-ui
 - https://github.com/longhorn/longhorn-tests
 - https://github.com/longhorn/backing-image-manager
-version: 1.2.200+up1.2.2
+version: 1.2.300-rc1+up1.2.3-rc1
```

---

Both these patches need to address @jiaqiluo's comments in https://github.com/rancher/charts/pull/1639#pullrequestreview-827104971, namely:
- rancher-version annotation that was removed needs to be reintroduced, this was added out-of-band by Donnie, which is why it wasn't reflected in your generated-changes on rebasing https://github.com/rancher/charts/commit/f43aaba87275d2881ffc702d8413563a5029ee49. Can you either add this in as a patch or introduce to your upstream chart, if necessary?
- We have now moved to using a rancher annotation based kube-version in addition / in replacement of the kubeVersion annotation used by Helm; since Longhorn already uses kubeVersion, can you just copy the contents of that field into the new `catalog.cattle.io/kube-version` field? Here is an example https://github.com/rancher/charts/blob/0f3c2b79003633f88358813e88369d594f9baa9c/charts/rancher-istio/rancher-istio/100.1.0%2Bup1.11.4/Chart.yaml#L4, it's pretty much the same value as the `kubeVersion` takes in today in the `Chart.yaml`. In the future, we will also need to keep those in sync.
- We also want to track the upstream version in `catalog.cattle.io/upstream-version`, since we're also planning to use that to facilitate upgrades in the Rancher UI (similar reasons to why we introduced the kube-version annotation); can you set that and make the value `1.2.3-rc1`?

Other than those Chart.yaml changes, this looks good to me.

**Comment by innobead** [comment]
> can you update the table in [this issue](https://github.com/rancher/rancher/issues/35100) to include Longhorn?

Updated only for 2.6.

**Review by jiaqiluo — APPROVED** [review]
LGTM

File: packages/longhorn-1.1/generated-changes/patch/Chart.yaml.patch [hunk 1]
```diff
@@ -13,3 +13,7 @@
 +  catalog.cattle.io/display-name: "Longhorn"
```
Comment (aiyengar2) [inline packages/longhorn-1.1/generated-changes/patch/Chart.yaml.patch:13]: Diff seems to show:

```diff
diff -uNr charts/longhorn-1.1/longhorn/1.1.200+up1.1.2/Chart.yaml charts/longhorn-1.1/longhorn/1.1.300-rc3+up1.1.3-rc3/Chart.yaml
--- charts/longhorn-1.1/longhorn/1.1.200+up1.1.2/Chart.yaml	2021-12-09 12:37:14.000000000 -0800
+++ charts/longhorn-1.1/longhorn/1.1.300-rc3+up1.1.3-rc3/Chart.yaml	2021-12-14 09:45:58.000000000 -0800
@@ -2,14 +2,16 @@
   catalog.cattle.io/auto-install: longhorn-crd=match
   catalog.cattle.io/certified: rancher
   catalog.cattle.io/display-name: Longhorn
+  catalog.cattle.io/kube-version: '>= v1.16.0-0, < v1.22.0-0'
   catalog.cattle.io/namespace: longhorn-system
-  catalog.cattle.io/rancher-version: <2.5.99-0
   catalog.cattle.io/os: linux
   catalog.cattle.io/provides-gvr: longhorn.io/v1beta1
+  catalog.cattle.io/rancher-version: <2.5.99-0
   catalog.cattle.io/release-name: longhorn
   catalog.cattle.io/ui-component: longhorn
+  catalog.cattle.io/upstream-version: 1.1.3-rc3
 apiVersion: v1
-appVersion: v1.1.2
+appVersion: v1.1.3-rc3
 description: Longhorn is a distributed block storage system for Kubernetes.
 home: https://github.com/longhorn/longhorn
 icon: https://raw.githubusercontent.com/cncf/artwork/master/projects/longhorn/icon/color/longhorn-icon-color.png
@@ -21,7 +23,7 @@
 - device
 - iscsi
 - nfs
-kubeVersion: '>=v1.16.0-r0'
+kubeVersion: '>= v1.16.0-0, < v1.22.0-0'
 maintainers:
 - email: maintainers@longhorn.io
   name: Longhorn maintainers
@@ -36,4 +38,4 @@
 - https://github.com/longhorn/longhorn-ui
 - https://github.com/longhorn/longhorn-tests
 - https://github.com/longhorn/backing-image-manager
-version: 1.1.200+up1.1.2
+version: 1.1.300-rc3+up1.1.3-rc3
```

w.r.t. setting the **Helm kubeVersion**, @innobead just to confirm; this particular chart version is **never** expected to be deployed (even in an officially unsupported fashion) within a k8s 1.15- or 1.22+ cluster?

If narrowing the scope of that kubeVersion is something that the longhorn team would ever want to do in the future, we should leave that constraint in the Rancher kube-version and drop it from the Helm kubeVersion flag. If not, this looks good as-is.

cc: @brendarearden

**Review by aiyengar2 — APPROVED** [review]
Ran the following script again:

```
arvindiyengar: ~/Rancher/charts/src/github.com/rancher/charts 
$ diff -uNr charts/longhorn-1.1/longhorn/1.1.200+up1.1.2 charts/longhorn-1.1/longhorn/1.1.300-rc3+up1.1.3-rc3/ > main-chart-1.1.diff
arvindiyengar: ~/Rancher/charts/src/github.com/rancher/charts 
$ diff -uNr charts/longhorn-1.1/longhorn-crd/1.1.200+up1.1.2 charts/longhorn-1.1/longhorn-crd/1.1.300-rc3+up1.1.3-rc3 > crd-chart-1.1.diff
arvindiyengar: ~/Rancher/charts/src/github.com/rancher/charts 
$ diff -uNr charts/longhorn-1.2/longhorn/1.2.200+up1.2.2 charts/longhorn-1.2/longhorn/1.2.300-rc2+up1.2.3-rc2 > main-chart-1-2.diff
arvindiyengar: ~/Rancher/charts/src/github.com/rancher/charts 
$ diff -uNr charts/longhorn-1.2/longhorn-crd/1.2.200+up1.2.2 charts/longhorn-1.2/longhorn-crd/1.2.300-rc2+up1.2.3-rc2 > crd-chart-1-2.diff
```

LGTM. Asking one question regarding kubeVersion (just a clarification) before this is ready to merge. If that requires changes, will re-review.

File: packages/longhorn-1.1/generated-changes/patch/Chart.yaml.patch (hunk 1 above)
Comment (aiyengar2) [reply packages/longhorn-1.1/generated-changes/patch/Chart.yaml.patch:13]: That makes sense to me; I'm expecting that to be the case, which is why I approved regardless

Comment (PhanLe1010) [reply packages/longhorn-1.1/generated-changes/patch/Chart.yaml.patch:13]: Yeah, Longhorn v1.1.3 only supports Kubernetes >= 1.16 and < 1.22 (not including 1.22)

Comment (PhanLe1010) [reply packages/longhorn-1.1/generated-changes/patch/Chart.yaml.patch:13]: > If narrowing the scope of that kubeVersion is something that the longhorn team would ever want to do in the future, we should leave that constraint in the Rancher kube-version and drop it from the Helm kubeVersion flag

@aiyengar2 Can you give more detail about which variable to set?

Comment (aiyengar2) [reply packages/longhorn-1.1/generated-changes/patch/Chart.yaml.patch:13]: I don't think any variable needs to be changed, based on your answer in https://github.com/rancher/charts/pull/1639#discussion_r768955985.

If there's no reason why we would ever want to make this **specific chart version** available to 1.15- or 1.22+, then the kubeVersion and kube-version can be the same.

**Comment by innobead** [comment]
After approvals, need QA (@khushboo-rancher ) to do the final verification.

**Comment by meldafrawi** [comment]
Validation on Rancher v2.5.11
Longhorn v1.1.3 & v1.2.3 install, uninstall, and upgrade.

**Review by aiyengar2 — APPROVED** [review]
Ran the following script:

```bash
mkdir -p 1.1.2
tar xvzf assets/longhorn-1.1/longhorn-1.1.200+up1.1.2.tgz -C 1.1.2

mkdir -p 1.1.3
tar xvzf assets/longhorn-1.1/longhorn-1.1.300+up1.1.3.tgz -C 1.1.3

diff -uNr 1.1.2 1.1.3 > diff.diff
```

```diff
diff -uNr 1.1.2/longhorn/Chart.yaml 1.1.3/longhorn/Chart.yaml
--- 1.1.2/longhorn/Chart.yaml	2021-10-15 12:10:33.000000000 -0700
+++ 1.1.3/longhorn/Chart.yaml	2021-12-16 00:20:51.000000000 -0800
@@ -2,14 +2,16 @@
   catalog.cattle.io/auto-install: longhorn-crd=match
   catalog.cattle.io/certified: rancher
   catalog.cattle.io/display-name: Longhorn
+  catalog.cattle.io/kube-version: '>= v1.16.0-0, < v1.22.0-0'
   catalog.cattle.io/namespace: longhorn-system
   catalog.cattle.io/os: linux
   catalog.cattle.io/provides-gvr: longhorn.io/v1beta1
   catalog.cattle.io/rancher-version: <2.5.99-0
   catalog.cattle.io/release-name: longhorn
   catalog.cattle.io/ui-component: longhorn
+  catalog.cattle.io/upstream-version: 1.1.3
 apiVersion: v1
-appVersion: v1.1.2
+appVersion: v1.1.3
 description: Longhorn is a distributed block storage system for Kubernetes.
 home: https://github.com/longhorn/longhorn
 icon: https://raw.githubusercontent.com/cncf/artwork/master/projects/longhorn/icon/color/longhorn-icon-color.png
@@ -21,7 +23,7 @@
 - device
 - iscsi
 - nfs
-kubeVersion: '>=v1.16.0-r0'
+kubeVersion: '>= v1.16.0-0, < v1.22.0-0'
 maintainers:
 - email: maintainers@longhorn.io
   name: Longhorn maintainers
@@ -36,4 +38,4 @@
 - https://github.com/longhorn/longhorn-ui
 - https://github.com/longhorn/longhorn-tests
 - https://github.com/longhorn/backing-image-manager
-version: 1.1.200+up1.1.2
+version: 1.1.300+up1.1.3
diff -uNr 1.1.2/longhorn/questions.yml 1.1.3/longhorn/questions.yml
--- 1.1.2/longhorn/questions.yml	2021-10-15 12:10:33.000000000 -0700
+++ 1.1.3/longhorn/questions.yml	2021-12-16 00:20:51.000000000 -0800
@@ -17,7 +17,7 @@
     label: Longhorn Manager Image Repository
     group: "Longhorn Images Settings"
   - variable: image.longhorn.manager.tag
-    default: v1.1.2
+    default: v1.1.3
     description: "Specify Longhorn Manager Image Tag"
     type: string
     label: Longhorn Manager Image Tag
@@ -29,7 +29,7 @@
     label: Longhorn Engine Image Repository
     group: "Longhorn Images Settings"
   - variable: image.longhorn.engine.tag
-    default: v1.1.2
+    default: v1.1.3
     description: "Specify Longhorn Engine Image Tag"
     type: string
     label: Longhorn Engine Image Tag
@@ -41,7 +41,7 @@
     label: Longhorn UI Image Repository
     group: "Longhorn Images Settings"
   - variable: image.longhorn.ui.tag
-    default: v1.1.2
+    default: v1.1.3
     description: "Specify Longhorn UI Image Tag"
     type: string
     label: Longhorn UI Image Tag
@@ -53,7 +53,7 @@
     label: Longhorn Instance Manager Image Repository
     group: "Longhorn Images Settings"
   - variable: image.longhorn.instanceManager.tag
-    default: v1_20210621
+    default: v1_20211210
     description: "Specify Longhorn Instance Manager Image Tag"
     type: string
     label: Longhorn Instance Manager Image Tag
@@ -65,7 +65,7 @@
     label: Longhorn Share Manager Image Repository
     group: "Longhorn Images Settings"
   - variable: image.longhorn.shareManager.tag
-    default: v1_20210416
+    default: v1_20210416_patch1
     description: "Specify Longhorn Share Manager Image Tag"
     type: string
     label: Longhorn Share Manager Image Tag
@@ -77,7 +77,7 @@
     label: Longhorn Backing Image Manager Image Repository
     group: "Longhorn Images Settings"
   - variable: image.longhorn.backingImageManager.tag
-    default: v1_20210422
+    default: v1_20210422_patch1
     description: "Specify Longhorn Backing Image Manager Image Tag"
     type: string
     label: Longhorn Backing Image Manager Image Tag
@@ -326,7 +326,7 @@
     default: "true"
   - variable: defaultSettings.replicaZoneSoftAntiAffinity
     label: Replica Zone Level Soft Anti-Affinity
-    description: "Allow scheduling new Replicas of Volume to the Nodes in the same Zone as existing healthy Replicas. Nodes don't belong to any Zone will be treated as in the same Zone. By default true."
+    description: "Allow scheduling new Replicas of Volume to the Nodes in the same Zone as existing healthy Replicas. Nodes don't belong to any Zone will be treated as in the same Zone. Notice that Longhorn relies on label `topology.kubernetes.io/zone=<Zone name of the node>` in the Kubernetes node object to identify the zone. By default true."
     group: "Longhorn Default Settings"
     type: boolean
     default: "true"
diff -uNr 1.1.2/longhorn/templates/services.yaml 1.1.3/longhorn/templates/services.yaml
--- 1.1.2/longhorn/templates/services.yaml	1969-12-31 16:00:00.000000000 -0800
+++ 1.1.3/longhorn/templates/services.yaml	2021-12-16 00:20:51.000000000 -0800
@@ -0,0 +1,24 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels: {{- include "longhorn.labels" . | nindent 4 }}
+  name: longhorn-engine-manager
+  namespace: longhorn-system
+spec:
+  clusterIP: None
+  selector:
+    longhorn.io/component: instance-manager
+    longhorn.io/instance-manager-type: engine
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels: {{- include "longhorn.labels" . | nindent 4 }}
+  name: longhorn-replica-manager
+  namespace: longhorn-system
+spec:
+  clusterIP: None
+  selector:
+    longhorn.io/component: instance-manager
+    longhorn.io/instance-manager-type: replica
diff -uNr 1.1.2/longhorn/values.yaml 1.1.3/longhorn/values.yaml
--- 1.1.2/longhorn/values.yaml	2021-10-15 12:10:33.000000000 -0700
+++ 1.1.3/longhorn/values.yaml	2021-12-16 00:20:51.000000000 -0800
@@ -9,22 +9,22 @@
   longhorn:
     engine:
       repository: rancher/mirrored-longhornio-longhorn-engine
-      tag: v1.1.2
+      tag: v1.1.3
     manager:
       repository: rancher/mirrored-longhornio-longhorn-manager
-      tag: v1.1.2
+      tag: v1.1.3
     ui:
       repository: rancher/mirrored-longhornio-longhorn-ui
-      tag: v1.1.2
+      tag: v1.1.3
     instanceManager:
       repository: rancher/mirrored-longhornio-longhorn-instance-manager
-      tag: v1_20210621
+      tag: v1_20211210
     shareManager:
       repository: rancher/mirrored-longhornio-longhorn-share-manager
-      tag: v1_20210416
+      tag: v1_20210416_patch1
     backingImageManager:
       repository: rancher/mirrored-longhornio-backing-image-manager
-      tag: v1_20210422
+      tag: v1_20210422_patch1
   csi:
     attacher:
       repository: rancher/mirrored-longhornio-csi-attacher
```

---

Ran the following script:

```bash
mkdir -p 1.1.2-crd
tar xvzf assets/longhorn-1.1/longhorn-crd-1.1.200+up1.1.2.tgz -C 1.1.2-crd

mkdir -p 1.1.3-crd
tar xvzf assets/longhorn-1.1/longhorn-crd-1.1.300+up1.1.3.tgz -C 1.1.3-crd

diff -uNr 1.1.2-crd 1.1.3-crd > diff.diff
```

```diff
diff -uNr 1.1.2-crd/longhorn-crd/Chart.yaml 1.1.3-crd/longhorn-crd/Chart.yaml
--- 1.1.2-crd/longhorn-crd/Chart.yaml	2021-09-09 11:10:40.000000000 -0700
+++ 1.1.3-crd/longhorn-crd/Chart.yaml	2021-12-16 00:20:51.000000000 -0800
@@ -7,4 +7,4 @@
 description: Installs the CRDs for longhorn.
 name: longhorn-crd
 type: application
-version: 1.1.200+up1.1.2
+version: 1.1.300+up1.1.3
diff -uNr 1.1.2-crd/longhorn-crd/templates/crds.yaml 1.1.3-crd/longhorn-crd/templates/crds.yaml
--- 1.1.2-crd/longhorn-crd/templates/crds.yaml	2021-09-09 11:10:40.000000000 -0700
+++ 1.1.3-crd/longhorn-crd/templates/crds.yaml	2021-12-16 00:20:51.000000000 -0800
@@ -1,12 +1,13 @@
+---
 apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   labels:
     app.kubernetes.io/name: longhorn
-    helm.sh/chart: longhorn-1.1.2
+    helm.sh/chart: longhorn-1.1.3
     app.kubernetes.io/managed-by: Helm
     app.kubernetes.io/instance: longhorn
-    app.kubernetes.io/version: v1.1.2
+    app.kubernetes.io/version: v1.1.3
     longhorn-manager: Engine
   name: engines.longhorn.io
 spec:
@@ -59,10 +60,10 @@
 metadata:
   labels:
     app.kubernetes.io/name: longhorn
-    helm.sh/chart: longhorn-1.1.2
+    helm.sh/chart: longhorn-1.1.3
     app.kubernetes.io/managed-by: Helm
     app.kubernetes.io/instance: longhorn
-    app.kubernetes.io/version: v1.1.2
+    app.kubernetes.io/version: v1.1.3
     longhorn-manager: Replica
   name: replicas.longhorn.io
 spec:
@@ -119,10 +120,10 @@
 metadata:
   labels:
     app.kubernetes.io/name: longhorn
-    helm.sh/chart: longhorn-1.1.2
+    helm.sh/chart: longhorn-1.1.3
     app.kubernetes.io/managed-by: Helm
     app.kubernetes.io/instance: longhorn
-    app.kubernetes.io/version: v1.1.2
+    app.kubernetes.io/version: v1.1.3
     longhorn-manager: Setting
   name: settings.longhorn.io
 spec:
@@ -156,10 +157,10 @@
 metadata:
   labels:
     app.kubernetes.io/name: longhorn
-    helm.sh/chart: longhorn-1.1.2
+    helm.sh/chart: longhorn-1.1.3
     app.kubernetes.io/managed-by: Helm
     app.kubernetes.io/instance: longhorn
-    app.kubernetes.io/version: v1.1.2
+    app.kubernetes.io/version: v1.1.3
     longhorn-manager: Volume
   name: volumes.longhorn.io
 spec:
@@ -216,10 +217,10 @@
 metadata:
   labels:
     app.kubernetes.io/name: longhorn
-    helm.sh/chart: longhorn-1.1.2
+    helm.sh/chart: longhorn-1.1.3
     app.kubernetes.io/managed-by: Helm
     app.kubernetes.io/instance: longhorn
-    app.kubernetes.io/version: v1.1.2
+    app.kubernetes.io/version: v1.1.3
     longhorn-manager: EngineImage
   name: engineimages.longhorn.io
 spec:
@@ -272,10 +273,10 @@
 metadata:
   labels:
     app.kubernetes.io/name: longhorn
-    helm.sh/chart: longhorn-1.1.2
+    helm.sh/chart: longhorn-1.1.3
     app.kubernetes.io/managed-by: Helm
     app.kubernetes.io/instance: longhorn
-    app.kubernetes.io/version: v1.1.2
+    app.kubernetes.io/version: v1.1.3
     longhorn-manager: Node
   name: nodes.longhorn.io
 spec:
@@ -324,10 +325,10 @@
 metadata:
   labels:
     app.kubernetes.io/name: longhorn
-    helm.sh/chart: longhorn-1.1.2
+    helm.sh/chart: longhorn-1.1.3
     app.kubernetes.io/managed-by: Helm
     app.kubernetes.io/instance: longhorn
-    app.kubernetes.io/version: v1.1.2
+    app.kubernetes.io/version: v1.1.3
     longhorn-manager: InstanceManager
   name: instancemanagers.longhorn.io
 spec:
@@ -376,10 +377,10 @@
 metadata:
   labels:
     app.kubernetes.io/name: longhorn
-    helm.sh/chart: longhorn-1.1.2
+    helm.sh/chart: longhorn-1.1.3
     app.kubernetes.io/managed-by: Helm
     app.kubernetes.io/instance: longhorn
-    app.kubernetes.io/version: v1.1.2
+    app.kubernetes.io/version: v1.1.3
     longhorn-manager: ShareManager
   name: sharemanagers.longhorn.io
 spec:
@@ -424,10 +425,10 @@
 metadata:
   labels:
     app.kubernetes.io/name: longhorn
-    helm.sh/chart: longhorn-1.1.2
+    helm.sh/chart: longhorn-1.1.3
     app.kubernetes.io/managed-by: Helm
     app.kubernetes.io/instance: longhorn
-    app.kubernetes.io/version: v1.1.2
+    app.kubernetes.io/version: v1.1.3
     longhorn-manager: BackingImage
   name: backingimages.longhorn.io
 spec:
@@ -437,7 +438,7 @@
     listKind: BackingImageList
     plural: backingimages
     shortNames:
-      - lhbi
+    - lhbi
     singular: backingimage
   scope: Namespaced
   versions:
@@ -468,10 +469,10 @@
 metadata:
   labels:
     app.kubernetes.io/name: longhorn
-    helm.sh/chart: longhorn-1.1.2
+    helm.sh/chart: longhorn-1.1.3
     app.kubernetes.io/managed-by: Helm
     app.kubernetes.io/instance: longhorn
-    app.kubernetes.io/version: v1.1.2
+    app.kubernetes.io/version: v1.1.3
     longhorn-manager: BackingImageManager
   name: backingimagemanagers.longhorn.io
 spec:
@@ -481,7 +482,7 @@
     listKind: BackingImageManagerList
     plural: backingimagemanagers
     shortNames:
-      - lhbim
+    - lhbim
     singular: backingimagemanager
   scope: Namespaced
   versions:
```

---

Ran the following script:

```bash
mkdir -p 1.2.2
tar xvzf assets/longhorn-1.2/longhorn-1.2.200+up1.2.2.tgz -C 1.2.2

mkdir -p 1.2.3
tar xvzf assets/longhorn-1.2/longhorn-1.2.300+up1.2.3.tgz -C 1.2.3

diff -uNr 1.2.2 1.2.3 > diff.diff
```

```diff
diff -uNr 1.2.2/longhorn/Chart.yaml 1.2.3/longhorn/Chart.yaml
--- 1.2.2/longhorn/Chart.yaml	2021-10-18 14:24:43.000000000 -0700
+++ 1.2.3/longhorn/Chart.yaml	2021-12-15 23:28:08.000000000 -0800
@@ -2,14 +2,16 @@
   catalog.cattle.io/auto-install: longhorn-crd=match
   catalog.cattle.io/certified: rancher
   catalog.cattle.io/display-name: Longhorn
+  catalog.cattle.io/kube-version: '>=1.18.0-0'
   catalog.cattle.io/namespace: longhorn-system
   catalog.cattle.io/os: linux
   catalog.cattle.io/provides-gvr: longhorn.io/v1beta1
   catalog.cattle.io/rancher-version: <2.5.99-0
   catalog.cattle.io/release-name: longhorn
   catalog.cattle.io/ui-component: longhorn
+  catalog.cattle.io/upstream-version: 1.2.3
 apiVersion: v1
-appVersion: v1.2.2
+appVersion: v1.2.3
 description: Longhorn is a distributed block storage system for Kubernetes.
 home: https://github.com/longhorn/longhorn
 icon: https://raw.githubusercontent.com/cncf/artwork/master/projects/longhorn/icon/color/longhorn-icon-color.png
@@ -35,4 +37,4 @@
 - https://github.com/longhorn/longhorn-ui
 - https://github.com/longhorn/longhorn-tests
 - https://github.com/longhorn/backing-image-manager
-version: 1.2.200+up1.2.2
+version: 1.2.300+up1.2.3
diff -uNr 1.2.2/longhorn/questions.yml 1.2.3/longhorn/questions.yml
--- 1.2.2/longhorn/questions.yml	2021-10-18 14:24:43.000000000 -0700
+++ 1.2.3/longhorn/questions.yml	2021-12-15 23:28:08.000000000 -0800
@@ -17,7 +17,7 @@
     label: Longhorn Manager Image Repository
     group: "Longhorn Images Settings"
   - variable: image.longhorn.manager.tag
-    default: v1.2.2
+    default: v1.2.3
     description: "Specify Longhorn Manager Image Tag"
     type: string
     label: Longhorn Manager Image Tag
@@ -29,7 +29,7 @@
     label: Longhorn Engine Image Repository
     group: "Longhorn Images Settings"
   - variable: image.longhorn.engine.tag
-    default: v1.2.2
+    default: v1.2.3
     description: "Specify Longhorn Engine Image Tag"
     type: string
     label: Longhorn Engine Image Tag
@@ -41,7 +41,7 @@
     label: Longhorn UI Image Repository
     group: "Longhorn Images Settings"
   - variable: image.longhorn.ui.tag
-    default: v1.2.2
+    default: v1.2.3
     description: "Specify Longhorn UI Image Tag"
     type: string
     label: Longhorn UI Image Tag
@@ -53,7 +53,7 @@
     label: Longhorn Instance Manager Image Repository
     group: "Longhorn Images Settings"
   - variable: image.longhorn.instanceManager.tag
-    default: v1_20210731
+    default: v1_20211210
     description: "Specify Longhorn Instance Manager Image Tag"
     type: string
     label: Longhorn Instance Manager Image Tag
@@ -65,7 +65,7 @@
     label: Longhorn Share Manager Image Repository
     group: "Longhorn Images Settings"
   - variable: image.longhorn.shareManager.tag
-    default: v1_20210914
+    default: v1_20211020
     description: "Specify Longhorn Share Manager Image Tag"
     type: string
     label: Longhorn Share Manager Image Tag
diff -uNr 1.2.2/longhorn/templates/services.yaml 1.2.3/longhorn/templates/services.yaml
--- 1.2.2/longhorn/templates/services.yaml	1969-12-31 16:00:00.000000000 -0800
+++ 1.2.3/longhorn/templates/services.yaml	2021-12-15 23:28:08.000000000 -0800
@@ -0,0 +1,24 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels: {{- include "longhorn.labels" . | nindent 4 }}
+  name: longhorn-engine-manager
+  namespace: longhorn-system
+spec:
+  clusterIP: None
+  selector:
+    longhorn.io/component: instance-manager
+    longhorn.io/instance-manager-type: engine
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels: {{- include "longhorn.labels" . | nindent 4 }}
+  name: longhorn-replica-manager
+  namespace: longhorn-system
+spec:
+  clusterIP: None
+  selector:
+    longhorn.io/component: instance-manager
+    longhorn.io/instance-manager-type: replica
diff -uNr 1.2.2/longhorn/values.yaml 1.2.3/longhorn/values.yaml
--- 1.2.2/longhorn/values.yaml	2021-10-18 14:24:43.000000000 -0700
+++ 1.2.3/longhorn/values.yaml	2021-12-15 23:28:08.000000000 -0800
@@ -9,19 +9,19 @@
   longhorn:
     engine:
       repository: rancher/mirrored-longhornio-longhorn-engine
-      tag: v1.2.2
+      tag: v1.2.3
     manager:
       repository: rancher/mirrored-longhornio-longhorn-manager
-      tag: v1.2.2
+      tag: v1.2.3
     ui:
       repository: rancher/mirrored-longhornio-longhorn-ui
-      tag: v1.2.2
+      tag: v1.2.3
     instanceManager:
       repository: rancher/mirrored-longhornio-longhorn-instance-manager
-      tag: v1_20210731
+      tag: v1_20211210
     shareManager:
       repository: rancher/mirrored-longhornio-longhorn-share-manager
-      tag: v1_20210914
+      tag: v1_20211020
     backingImageManager:
       repository: rancher/mirrored-longhornio-backing-image-manager
       tag: v2_20210820
```

---

Ran the following script:

```bash
mkdir -p 1.2.2-crd
tar xvzf assets/longhorn-1.2/longhorn-crd-1.2.200+up1.2.2.tgz -C 1.2.2-crd

mkdir -p 1.2.3-crd
tar xvzf assets/longhorn-1.2/longhorn-crd-1.2.300+up1.2.3.tgz -C 1.2.3-crd

diff -uNr 1.2.2-crd 1.2.3-crd > diff.diff
```

```diff
diff -uNr 1.2.2-crd/longhorn-crd/Chart.yaml 1.2.3-crd/longhorn-crd/Chart.yaml
--- 1.2.2-crd/longhorn-crd/Chart.yaml	2021-10-07 05:33:53.000000000 -0700
+++ 1.2.3-crd/longhorn-crd/Chart.yaml	2021-12-15 23:28:08.000000000 -0800
@@ -7,4 +7,4 @@
 description: Installs the CRDs for longhorn.
 name: longhorn-crd
 type: application
-version: 1.2.200+up1.2.2
+version: 1.2.300+up1.2.3
```

---

Upstream version and kubeVersion added. No other issues noted.

LGTM

**Review by brendarearden — APPROVED** [review]
lgtm!

**Comment by innobead** [comment]
When to release?

**Comment by jehof** [comment]
When will this be released?

**Comment by brendarearden** [comment]
@jehof This will be bundled in the rancher 2.5.12 - if you want to checkout something early you can add the dev-v2.5 branch as a repository and deploy from there.

**Pull request merged**